lib/internal/watch_mode/files_watcher.js

```diff
@@ -74,17 +75,19 @@ class FilesWatcher extends EventEmitter {
   }
 
   #onChange(trigger) {
-    if (this.#debouncing.has(trigger)) {
-      return;
-    }
     if (this.#mode === 'filter' && !this.#filteredFiles.has(trigger)) {
       return;
     }
-    this.#debouncing.add(trigger);
+
     const owners = this.#depencencyOwners.get(trigger);
-    setTimeout(() => {
-      this.#debouncing.delete(trigger);
+    if (owners) for (const o of owners) this.#debounceOwners.add(o);
+
+    clearTimeout(this.#debounceTimer);
+    this.#debounceTimer = setTimeout(() => {
+      this.#debounceTimer = null;
+      const owners = this.#debounceOwners;
       this.emit('changed', { owners });
+      this.#debounceOwners = new SafeSet();
```

Comment:
```suggestion
      this.#debounceOwners.clear();
```

Reply:
The reason I did that is to allow async processing of the `owners` by the listeners. I figured it would be more efficient to swap the `Set` rather than doing:
```ts
      const owners = new Set(this.#debounceOwners);
      this.emit('changed', { owners });
      this.#debounceOwners.clear();
```

What do you think ?

Reply:
If you think that async processing of the event is not something we should care about here, then I can just implement your change.